### PR TITLE
fix(planner): apply step_size multiplier in ProphetPredictor.predict_next()

### LINKED ISSUE #7211: [BUG]: ProphetPredictor.predict_next() uses wrong forecast timestamp when step_size > 1
### Describe the Bug

`ProphetPredictor.predict_next()` computes the future timestamp passed to Prophet as:

```python
next_timestamp = self.start_date + timedelta(seconds=self.curr_step)
```

But `add_data_point()` stores each training sample's timestamp as:

```python
timestamp = self.start_date + timedelta(seconds=self.curr_step * self.step_size)
```

The `* self.step_size` factor is missing in `predict_next()`. Because `step_size` defaults to 180 s (the `throughput_adjustment_interval`), the forecast target falls ~180× too early relative to the training data, causing Prophet to extrapolate into the past instead of one step into the future.

### Steps to Reproduce

```python
from unittest.mock import MagicMock
from dynamo.planner.utils.load_predictor import ProphetPredictor

cfg = MagicMock()
cfg.load_predictor_log1p = False
cfg.throughput_adjustment_interval = 180  # default step_size
cfg.prophet_window_size = 50

predictor = ProphetPredictor(cfg)
for v in [10, 12, 11, 13, 10, 9]:
    predictor.add_data_point(float(v))

# After 6 points, curr_step == 6
# Last training timestamp : start + 5 * 180 = start + 900 s
# Expected next timestamp : start + 6 * 180 = start + 1080 s  ✅
# Actual next timestamp   : start + 6       = start + 6 s     ❌
```

### Expected Behavior

The timestamp passed to `Prophet.predict()` should be exactly one `step_size` ahead of the last training sample:

```python
next_timestamp = self.start_date + timedelta(seconds=self.curr_step * self.step_size)
```

This keeps the forecast point temporally consistent with the training timeline so Prophet can extrapolate correctly.

### Actual Behavior

With the default `step_size = 180`, after 6 data points the forecast target is `start + 6 s` instead of `start + 1080 s`. Prophet therefore predicts a value far in the past of its own training window, producing inaccurate load forecasts.

### Environment

- **Component**: `components/src/dynamo/planner/utils/load_predictor.py`
- **Class / method**: `ProphetPredictor.predict_next()`
- **Triggered when**: `throughput_adjustment_interval` (step_size) > 1 — always true in production (default = 180 s)

### Additional Context

Fix and regression tests are available in PR #7208:
https://github.com/ai-dynamo/dynamo/pull/7208

One-line fix:

```diff
- next_timestamp = self.start_date + timedelta(seconds=self.curr_step)
+ next_timestamp = self.start_date + timedelta(seconds=self.curr_step * self.step_size)
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by AsadShahid04**
## Summary

Fixes #7211

In `ProphetPredictor`, `add_data_point()` stores timestamps as:
```python
timestamp = self.start_date + timedelta(seconds=self.curr_step * self.step_size)
```

But `predict_next()` was computing the forecast target as:
```python
next_timestamp = self.start_date + timedelta(seconds=self.curr_step)
```

The `* self.step_size` factor was missing. With the default `step_size=180`, after 6 data points the forecast target was `start+6s` instead of `start+1080s` — Prophet was being asked to extrapolate **into the past** rather than into the future, producing nonsensical predictions.

## Fix

One-line change: add the missing `* self.step_size` multiplier so the forecast timestamp correctly lands one interval ahead of the last observed point.

```python
# Before (bug)
next_timestamp = self.start_date + timedelta(seconds=self.curr_step)

# After (fix)
next_timestamp = self.start_date + timedelta(seconds=self.curr_step * self.step_size)
```

## Tests

Added `tests/planner/unit/test_prophet_predictor.py` covering:
- Timestamp alignment between `add_data_point()` and `predict_next()` for `step_size=1` and `step_size=180`
- Forecast target is strictly after all stored timestamps (not in the past)
- `predict_next()` returns a non-negative float end-to-end
- Fallback to last value when below `minimum_data_points`

## Strategic context

`ProphetPredictor` is part of the Planner's autoscaling pipeline — a core component under the Production Grade Serving roadmap pillar. A broken timestamp meant Prophet's forecasts were unreliable for any `step_size > 1`, silently degrading autoscaling quality.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed timestamp alignment in forecasting logic to properly scale timestamps by time-step size, ensuring consistent timing between buffered training data and prediction generation.

* **Tests**
  * Added comprehensive unit tests validating timestamp alignment between data collection and forecast generation, including edge cases with minimal data and various time-step intervals.

### CODE REVIEW

File: tests/planner/unit/test_prophet_predictor.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+"""Unit tests for ProphetPredictor — focusing on timestamp alignment between
+add_data_point() and predict_next().
+
+Regression test for the bug where predict_next() computed:
+    next_timestamp = start_date + timedelta(seconds=curr_step)
+instead of:
+    next_timestamp = start_date + timedelta(seconds=curr_step * step_size)
+
+With the default step_size=180 this meant the forecast target was 180× too
+early — Prophet was asked to extrapolate into the past, producing garbage.
```
[the remark this answers is not in the document]
Comment (AsadShahid04) [reply tests/planner/unit/test_prophet_predictor.py]: This has been addressed in the PR. The docstring in the submitted test file uses '180x' (ASCII) rather than the Unicode multiplication sign '×', satisfying Ruff RUF002.

File: tests/planner/unit/test_prophet_predictor.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            )
+
+        # The forecast target must be exactly one step_size ahead of the last stored point.
+        actual_gap = expected_next_ts - last_stored_ts
+        assert actual_gap == expected_step_delta, (
+            f"step_size={step_size}: forecast target is {actual_gap.total_seconds()}s "
+            f"ahead of last stored timestamp, expected {step_size}s. "
+            "This indicates the * self.step_size multiplier is missing in predict_next()."
+        )
+
```
[the remark this answers is not in the document]
Comment (AsadShahid04) [reply tests/planner/unit/test_prophet_predictor.py:99]: This has been addressed in the PR. The test `test_forecast_target_is_one_step_ahead` uses `patch('dynamo.planner.utils.load_predictor.Prophet')` to intercept the actual `future_df` argument passed to `Prophet.predict()`, then directly asserts the `ds` timestamp value. A regression in `predict_next()` that drops the `step_size` multiplier would cause this test to fail.

File: tests/planner/unit/test_prophet_predictor.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        """Feed n_points into the predictor and return the list of stored ds values."""
+        for _ in range(n_points):
+            predictor.add_data_point(value)
+        return [entry["ds"] for entry in predictor.data_buffer]
+
+    @pytest.mark.parametrize("step_size", [1, 180])
+    def test_forecast_target_is_one_step_ahead(self, step_size):
+        """The forecast timestamp must equal the last stored timestamp + step_size seconds."""
+        from dynamo.planner.utils.load_predictor import ProphetPredictor
+
```
[the remark this answers is not in the document]
Comment (AsadShahid04) [reply tests/planner/unit/test_prophet_predictor.py]: This has been addressed in the PR. All imports (`timedelta`, `MagicMock`, `patch`, `pandas`, `pytest`, `ProphetPredictor`) are placed at module scope (lines 27-33 in the submitted file), not inside functions or classes.

**Comment by aylonsk** [comment]
The underlying issue #7211 was closed as completed on 2026-05-07, and `predict_next` on main
already applies the `step_size` factor (`planner/core/load/predictors.py:306-307`). The predictors
have also moved out of `planner/utils/load_predictor.py`, so this no longer rebases. I think this can be safely closed.

**Comment by rmccorm4** [comment]
Closing as superseded by https://github.com/ai-dynamo/dynamo/pull/7208. Please feel free to open a new PR if this is still needed. CC @aylonsk
